character set: include useful warning (#3730)
pingcap/docs: character-set-and-collation.md

5 rows in set (0.01 sec)
```

> **Warning:**
>

> > TiDB incorrectly treats latin1 as a subset of utf8. This can lead to unexpected behaviors when storing characters which differ between latin1 and utf8 encodings. It is strongly recommended to the utf8mb4 character set. See [TiDB #18955](https://github.com/pingcap/tidb/issues/18955) for additional details.

Reviewer: ```suggestion
> TiDB incorrectly treats latin1 as a subset of utf8. This can lead to unexpected behaviors when you store characters that differ between latin1 and utf8 encodings. It is strongly recommended to the utf8mb4 character set. See [TiDB #18955](https://github.com/pingcap/tidb/issues/18955) for more details.
```
